provider/core/src/datagen/iter.rs

```diff
@@ -5,37 +5,32 @@
 //! Collection of iteration APIs for data providers.
 
 use crate::prelude::*;
+use icu_locid::Locale;
 
-/// A [`DynProvider`] that can iterate over all supported [`ResourceOptions`] for a certain key.
+/// A [`DynProvider`] that can iterate over all supported [`Locale`]s for a certain key.
 ///
 /// Implementing this trait means that a data provider knows all of the data it can successfully
 /// return from a load request.
 pub trait IterableDynProvider<M: DataMarker>: DynProvider<M> {
-    /// Given a [`ResourceKey`], returns a list of [`ResourceOptions`].
-    fn supported_options_for_key(
-        &self,
-        key: ResourceKey,
-    ) -> Result<Vec<ResourceOptions>, DataError>;
+    /// Given a [`ResourceKey`], returns a list of [`Locale`]s.
+    fn supported_locales_for_key(&self, key: ResourceKey) -> Result<Vec<Locale>, DataError>;
```

Comment:
Issue: We may want to use the name `supported_locales_for_key` for the actual supported-locales logic proposed in https://github.com/unicode-org/icu4x/issues/58#issuecomment-1163470414 in the post-1.x timeframe. I would like this trait to remain datagen-specific for the time being.

The other issue here is that I think we may want to retire this mechanism in the data exporter entirely, and instead use a declarative list of locales combined with locale fallback. At least when reading CLDR data, we only need to load the list of CLDR locales one time, and then use the same list for all of the data keys. See https://github.com/unicode-org/icu4x/issues/834 for further discussion.